core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3105,4 +3281,182 @@ public Long getValue() {
+    private String mIp;
+    private Inode mSrcInode;
+
+    @Override
+    public MasterAuditContext setAllowed(boolean allowed) {
+      mAllowed = allowed;
+      return this;
+    }
+
+    @Override
+    public MasterAuditContext setSuccess(boolean success) {
+      mSuccess = success;
+      return this;
+    }
+
+    /**
+     * Sets mCommand field.
+     *
+     * @param command the command associated with this {@link alluxio.master.Master}
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setCommand(String command) {
+      mCommand = command;
+      return this;
+    }
+
+    /**
+     * Sets mSrcPath field.
+     *
+     * @param srcPath the source path of the command
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setSrcPath(AlluxioURI srcPath) {
+      mSrcPath = srcPath;
+      return this;
+    }
+
+    /**
+     * Sets mDstPath field.
+     *
+     * @param dstPath the destination path of the command
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setDstPath(AlluxioURI dstPath) {
+      mDstPath = dstPath;
+      return this;
+    }
+
+    /**
+     * Sets mUgi field.
+     *
+     * @param ugi the client user name of the authenticated client user of this thread
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setUgi(String ugi) {
+      mUgi = ugi;
+      return this;
+    }
+
+    /**
+     * Sets mAuthType field.
+     *
+     * @param authType the authentication type
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setAuthType(AuthType authType) {
+      mAuthType = authType;
+      return this;
+    }
+
+    /**
+     * Sets mIp field.
+     *
+     * @param ip the IP of the client
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setIp(String ip) {
+      mIp = ip;
+      return this;
+    }
+
+    /**
+     * Sets mSrcInode field.
+     *
+     * @param srcInode the source inode of this operation
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setSrcInode(Inode srcInode) {
+      mSrcInode = srcInode;
+      return this;
+    }
+
+    /**
+     * Constructor of {@link MasterAuditContext}.
+     *
+     * @param asyncAuditLogWriter
+     */
+    private MasterAuditContext(AsyncUserAccessAuditLogWriter asyncAuditLogWriter) {
+      mAsyncAuditLogWriter = asyncAuditLogWriter;
+      mAllowed = true;
+    }
+
+    @Override
+    public void close() {
+      if (mAsyncAuditLogWriter == null) {
+        return;
+      }
+      mAsyncAuditLogWriter.append(this);
+    }
+
+    @Override
+    public String toString() {
+      if (mSrcInode != null) {
+        short mode = mSrcInode.getMode();
+        return String.format("allowed=%b\tugi=%s (AUTH=%s)\tip=%s\tcmd=%s\tsrc=%s\tdst=%s\t"
+                + "perm=%s:%s:%s%s%s\tsuccess=%b",
```

Comment:
Done. 
I think we should put succeeded field either at the beginning or at the end of the message. Since it also describes the status, then I guess we should put it at the beginning before "allowed".